src/LogicTasks/Syntax/TreeToFormula.hs

```diff
@@ -84,21 +84,27 @@ partialGrade _ sol
 
 completeGrade :: (OutputMonad m, MonadIO m) => FilePath -> TreeToFormulaInst -> TreeFormulaAnswer -> LangM m
 completeGrade path inst sol
-    | treeAnswer /= tree inst = refuse $ do
+    | treeAnswer /= correctTree = refuse $ do
         instruct $ do
           english "Your solution is not correct. The syntax tree for your entered formula looks like this:"
           german "Ihre Abgabe ist nicht die korrekte Lösung. Der Syntaxbaum zu Ihrer eingegebenen Formel sieht so aus:"
 
         image $=<< liftIO $ cacheTree (transferToPicture treeAnswer) path
 
-        when (addExtraHintsOnSemanticEquivalence inst && isSemanticEqual treeAnswer (tree inst)) $
+        when (showSolution inst) $
+          example (show (display correctTree)) $ do
+            english "A possible solution for this task is:"
+            german "Eine mögliche Lösung für die Aufgabe ist:"
+
+        when (addExtraHintsOnSemanticEquivalence inst && isSemanticEqual treeAnswer correctTree) $
```

Comment:
The order here is somewhat weird. If both the `when`s above will be true, then in the output first a tree is shown ("The syntax tree for your entered formula looks like this:"), then a formula is shown (`show (display correctTree)`), then reference is made to the *syntax tree* ("This syntax tree is semantically equivalent to the original one, but not identical.")

It seems it would make more sense to first show the wrong tree, then say it is not identical to the intended one, then say what the correct formula is.

Reply:
(I know that this is not caused by this pull request here, but by some other merge earlier on in some pull request or directly on `master`.)